src/rules/indentRule.ts

```diff
@@ -121,6 +156,383 @@ function walk(ctx: Lint.WalkContext<Options>): void {
+            && !isSameLine(root, parent.getStart(), nodePos)) {
+                result ++;
+        }
+        if (inStatementParen(parent)) {
+            if (!leadingWithBinaryOperator(node) && !isElseIf(parent) && !tokenInBinary) {
+                result ++;
+            }
+        } else if (isInParen(parent, root)) {
+            if (!isSameLineWithParent(parent, root)) {
+                result ++;
+            }
+        }
+
+        if (sameLineWithBinary && parent.kind !== ts.SyntaxKind.SourceFile) {
+            if (parent.kind === ts.SyntaxKind.VariableDeclaration) {
+                result ++;
+            }
+        }
+        parent = parent.parent;
+    }
+
+    return result;
+}
+
+// function isSameLineWithConditionalExpr(sourceFile: ts.SourceFile, nodePos: number, parent: ts.Node): boolean {
+//     if (parent.kind !== ts.SyntaxKind.ConditionalExpression && parent.kind !== ts.SyntaxKind.ClassDeclaration) {
+//         return false;
+//     }
+//     return isSameLine(sourceFile, nodePos, parent.getStart());
+// }
+function isCaseWithBlock(node: ts.Node): boolean {
+    if (node.parent === undefined) {
+        return false;
+    }
+    return node.kind === ts.SyntaxKind.Block
+         && node.parent.kind === ts.SyntaxKind.CaseClause;
+}
+
+function isChainCall(node: ts.Node): boolean {
+    if (node.parent === undefined) {
+        return false;
+    }
+    if (node.kind !== ts.SyntaxKind.DotToken) {
+        return false;
+    }
+
+    const parent: ts.Node = node.parent;
+    let result = false;
+    ts.forEachChild(parent, (child: ts.Node) => {
+        if (child.kind === ts.SyntaxKind.CallExpression) {
+            result = true;
+        }
+    });
+    return result;
+}
+
+function getChainableCallNumberBeforeNode(sourceFile: ts.SourceFile, nodePos: number): number {
+    let result = 0;
+    const checkedNodes: ts.Node[] = [];
+    for (const { pos } of getLineRanges(sourceFile)) {
+        if (pos >= nodePos) {
+            continue;
+        }
+        const currentNode: ts.Node | undefined = getTokenAtPosition(sourceFile, pos);
+        if (currentNode === undefined) {
+            continue;
+        }
+        if (checkedNodes.indexOf(currentNode) > -1) {
+            continue;
+        } else {
+            checkedNodes.push(currentNode);
+        }
+        if (isChainCall(currentNode)) {
+            if (currentNode.parent === undefined) {
+                continue;
+            }
+            if (currentNode.parent.parent === undefined) {
+                continue;
+            }
+            const grandpa: ts.Node = currentNode.parent.parent;
+            if (grandpa.getEnd() > nodePos) {
+                result++;
+            }
+        }
+    }
+    return result;
+}
+/*
+* Check for property access (includes chaining function call) just like this:
+foo()
+.then()
+In this case, `.then` should have one more indent.
+Maybe this should depend on a configuration.
```

Comment:
use proper JSDoc formatting please

Reply:
also, is this last line a question for code reviewers? I don't think this is a helpful comment without more elaboration. Either remove it or make the "maybe" part more clear.

Reply:
done